probe-rs-tools/src/bin/probe-rs/cmd/cargo_embed/rttui/tab.rs

```diff
@@ -107,23 +106,18 @@ impl Tab {
     pub fn send_input(
         &mut self,
         core: &mut Core,
-        channels: &mut BTreeMap<usize, RttActiveDownChannel>,
+        channels: &mut [RttActiveDownChannel],
     ) -> anyhow::Result<()> {
         if let Some((channel, input)) = self.down_channel.as_mut() {
-            let channel = channels.get_mut(channel).expect("down channel disappeared");
             input.push('\n');
-            channel.push_rtt(core, input.as_str())?;
+            channels[*channel].push_rtt(core, input.as_str())?;
```

Comment:
I don't like this :D I liked the (albeit not very descriptive) old message better :)

Reply:
For something that is impossible, I prefer the simpler syntax.

Reply:
It's also important to note I think that the old `expect` did not include anything about the key, while now the panic message would be `index out of bounds: the len is X but the index is Y`. The alternative would be `.expect_with(|| panic!("...", channel, channels.len())` which is... verbose.

Reply:
That being said, I've been toying with the idea of just putting the channel into an `Rc` or `Arc` and remove this indexing stuff, so this may not be the final form.

Reply:
Ok, fair, I was not considering that it cannot actually occur really.